src/GoTrueClient.ts

```diff
@@ -1069,9 +1071,9 @@ export default class GoTrueClient {
       } = await this.getSession()
       if (error) throw error
 
-      this.stateChangeEmitters.get(id)?.callback('INITIAL_SESSION', session)
+      await this.stateChangeEmitters.get(id)?.callback('INITIAL_SESSION', session)
     } catch (err) {
-      this.stateChangeEmitters.get(id)?.callback('INITIAL_SESSION', null)
+      await this.stateChangeEmitters.get(id)?.callback('INITIAL_SESSION', null)
```

Comment:
I don't think these should be `await`-ed.

Reply:
bumping this - am keen to know why these statements shouldn't be `await`-ed as well

Reply:
Trying to remember. 😂

Reply:
Ah! Nevermind. I thought there was the possibility of recursive await, but doesn't appear so.